src/components/OrdersWidget/index.tsx

```diff
@@ -83,15 +88,41 @@ function classifyOrders(
   })
 }
 
-const compareFnFactory = (topic: TopicNames, asc: boolean) => (lhs: AuctionElement, rhs: AuctionElement): number => {
+function checkTokenAgainstSearch(token: TokenDetails | null, searchText: string): boolean {
+  if (!token) return false
+  return (
+    token?.symbol?.toLowerCase().includes(searchText) ||
+    token?.name?.toLowerCase().includes(searchText) ||
+    token?.address.toLowerCase().includes(searchText)
+  )
+}
+
+const filterOrdersFn = (searchTxt: string) => ({ id, buyToken, sellToken }: DetailedAuctionElement): boolean => {
+  if (searchTxt === '') return true
+
+  return (
+    !!id.includes(searchTxt) ||
+    checkTokenAgainstSearch(buyToken as TokenDetails, searchTxt) ||
+    checkTokenAgainstSearch(sellToken as TokenDetails, searchTxt)
```

Comment:
Would
```suggestion
    (buyToken && checkTokenAgainstSearch(buyToken, searchTxt)) ||
    (sellToken && checkTokenAgainstSearch(sellToken, searchTxt))
```
work? Eliminating type casting